### DISCUSSION TITLE
Using specific !important class with global config important

### DISCUSSION THREAD

**Discussion by irg1008** [tailwindlabs/tailwindcss#12511] [Help] [answered]
I am trying to use `!` with tailwindcss classes inside a component while having `important: true` in `taiwlind.config.js` and it won't work. 
For example if I have a button with class "bg-red-500 !bg-blue-500" the applied class is "bg-red-500" even though the latter is marked as important.

Is there a way to set global important config to true while respecting component-specific important rules?

**Comment by wongjn** [comment, marked as answer]
`important: true` adds `!important` to utility declarations. Thus, both `bg-red-500` and `!bg-blue-500` both have the same specificity since they would both have `!important` in their declarations. This makes the `!` modifier pointless. `bg-red-500` is applied since it would appear later in the generated CSS.

To work around this, you could consider:
- Restructuring your component so that only one `bg-<color>` class is ever applied at any one time, like:
  ```jsx
  function Component({ bg = 'bg-red-500' }) {
    return <div class={bg} />
  }
  
  function App() {
    return (
      <>
        {/* Will be red */}
        <Component />
        {/* Will be blue */}
        <Component bg="bg-blue-500" />
      </>
    );
  }
  ```
- Increase the specificity of the class to take precedence arbitrarily, such as by using an attribute selector which adds `0 1 0` specificity:
  ```html
  <div class="bg-red-500 !bg-blue-500">Will be red</div>
  <div class="bg-red-500 [&[class]]:bg-blue-500">Will be blue</div>
  ```
- Use a different value for `important`. This would mean `!` would have an effect.
